src/lib.rs

```diff
@@ -160,6 +173,72 @@ impl L402Module {
             }
         }
     }
+
+    pub async fn verify_cashu_token(&self, token: &str, amount_msat: i64) -> Result<bool, String> {
+        // Decode the token from string
+        let token_decoded = match cdk::nuts::Token::from_str(token) {
+            Ok(token) => token,
+            Err(e) => {
+                eprintln!("Failed to decode Cashu token: {}", e);
+                return Err(format!("Failed to decode Cashu token: {}", e));
+            }
+        };
+        
+        // Check if the token is valid
+        if token_decoded.proofs().is_empty() {
+            return Ok(false);
+        }
+        
+        // Calculate total token amount in millisatoshis
+        let total_amount_msat: u64 = token_decoded.proofs().iter()
+            .map(|p| u64::from(p.amount) * MSAT_PER_SAT)
+            .sum();
+        
+        // Check if the token amount is sufficient
+        if total_amount_msat < amount_msat as u64 {
+            eprintln!("Cashu token amount insufficient: {} msat (required: {} msat)", 
+                total_amount_msat, amount_msat);
+            return Ok(false);
+        }
+        
+        println!("Successfully decoded Cashu token with {} proofs and {} msat (required: {} msat)", 
+            token_decoded.proofs().len(),
+            total_amount_msat,
+            amount_msat);
+        
+        // Extract mint URL from the token
+        let mint_url = token_decoded.mint_url()
+            .map_err(|e| format!("Failed to get mint URL: {}", e))?;
+        
+        // amount from total millisatoshis into satoshis
+        let amount = Some(cdk::Amount::from((total_amount_msat as u64) / MSAT_PER_SAT));
+
+        let unit = token_decoded.unit().unwrap();
+        
+        // Use the shared database instance
+        let db = CASHU_DB.get()
+            .ok_or_else(|| "Cashu database not initialized".to_string())?;
+            
+        let seed = rand::thread_rng().gen::<[u8; 32]>();
+        let wallet = cdk::wallet::Wallet::new(&mint_url.to_string(), unit, db.clone(), &seed, None)
+            .map_err(|e| format!("Failed to create wallet: {}", e))?;
+        
+        // Create default spending conditions and split target
+        let spending_conditions = None;
+        let amount_split_target = cdk::amount::SplitTarget::default();
+        let include_fees = false;
+
+        match wallet.swap(amount, amount_split_target, token_decoded.proofs(), spending_conditions, include_fees).await {
+            Ok(_) => {
+                eprintln!("Cashu token swap successful");
+                Ok(true)
+            },
+            Err(e) => {
+                eprintln!("Cashu token swap failed: {}", e);
+                Ok(false)
+            }
+        }
+    }

```

Comment:
You should use `wallet.receive` here https://docs.rs/cdk/latest/cdk/wallet/struct.Wallet.html#method.receive. Since you are using swap and passing in the amount the proofs are being set to reserved in the wallet, however you are not doing anything with them here so when you go to use the wallet later you will have `Reserved` proofs that should be unspent.